pancancer-arch-3/src/main/java/info/pancancer/arch3/worker/WorkerRunnable.java

```diff
@@ -452,10 +454,23 @@ public InetAddress getFirstNonLoopbackAddress() throws SocketException {
     private void finishJob(String message) {
         log.info("Publishing worker results to results channel " + this.resultsQueueName + ": " + message);
         try {
-            resultsChannel.basicPublish(this.resultsQueueName, this.resultsQueueName, MessageProperties.PERSISTENT_TEXT_PLAIN,
-                    message.getBytes(StandardCharsets.UTF_8));
-        } catch (IOException e) {
+            boolean success = false;
+            do {
+                try {
+                    resultsChannel.basicPublish(this.resultsQueueName, this.resultsQueueName, MessageProperties.PERSISTENT_TEXT_PLAIN,
+                            message.getBytes(StandardCharsets.UTF_8));
+                    resultsChannel.waitForConfirms();
+                    success = true;
+                } catch (AlreadyClosedException e) {
+                    // retry indefinitely if the connection is down
+                    log.error("could not send closed message, retrying", e);
```

Comment:
Maybe change error message to something clearer: "Could not send message: channel is closed. Retrying in 60 seconds..." ?